experimental/single_number_formatter/src/provider.rs

```diff
@@ -0,0 +1,79 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use tinystr::UnvalidatedTinyAsciiStr;
+use zerovec::{VarZeroVec, ZeroMap};
+
+#[icu_provider::data_struct(CurrencyEssentialsV1Marker = "currency/essentials@1")]
+#[derive(Default, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_singlenumberformatter::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyEssentialsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub currency_patterns_map: ZeroMap<'data, UnvalidatedTinyAsciiStr<3>, CurrencyPatterns>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub standard: Cow<'data, str>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub standard_alpha_next_to_number: Cow<'data, str>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub place_holders: VarZeroVec<'data, str>,
+}
+
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+#[repr(u8)]
+pub enum PatternSelection {
```

Comment:
Suggestion: you should be able to use `make_ule` on this enum and then use that ULE down below instead of making the `type PatternSelectionMask = u8;` typedef.

Reply:
I have tried it but `#[zerovec::make_ule(CurrencyPatternsULE)]` fails, see my comment in the code below

Reply:
ok, I don't know what error you were getting but we should follow up to fix it. Robert should be able to help.